store/dhash/dhash.go

```diff
@@ -62,9 +72,11 @@ func EncryptAES(payload, passphrase []byte) ([]byte, []byte, error) {
 	derivedKey := deriveKey([]byte(passphrase))
 
 	// Create initialization vector (nonse) to be used during encryption
-	// Nonce is derived from the mulithash (passpharase) so that encrypted payloads
-	// for the same multihash can be compared to each other without having to decrypt
-	nonce := SHA256(passphrase, nil)[:nonceLen]
+	// Nonce is derived from the passphrase concatenated with the payload so that the encrypted payloads
+	// for the same multihash can be compared to each other without having to decrypt them, as it's not possible.
+	payloadLen := make([]byte, 8)
```

Comment:
Concerning the payloadLen, the exact format isn't defined yet in https://github.com/ipfs/specs/pull/373.

Would it make more sense to use a constant byte array (e.g `make([]byte, 8)`) or to use an unsigned varint. As we mostly encrypt peerids, that are quite short, we don't always need a big number for the payload len. A varint allows to use less bytes to describe the payload len, and the maximal payload len can also be very large if needed.

I think that I am more in favor of an unsigned varint format for the payload len, as it gives more agility. WDYT?

cc: @ischasny @Jorropo @masih

Reply:
> we don't always need a big number for the payload len

These values are hashed over, so `payloadLen` isn't going to be a part of the encrypted value, i.e. see  line 79: `nonce := sha256Multiple(nil, noncePrefix, payloadLen, payload, passphrase)[:nonceLen]`. 

Does that make sense?

Reply:
Alright, all good! 👍🏻